src/test/java/edu/harvard/iq/dataverse/api/ThumbnailsIT.java

```diff
@@ -35,4 +42,65 @@ public void testDatasetThumbnail() {
+        String pathToJsonFile = "src/test/resources/json/complete-dataset-with-files.json";
+        Response createDatasetResponse = UtilIT.createDatasetViaNativeApi(dataverseAlias, pathToJsonFile, apiToken);
+        createDatasetResponse.prettyPrint();
+        Integer datasetId = UtilIT.getDatasetIdFromResponse(createDatasetResponse);
+
+        Response datasetAsJson = UtilIT.nativeGet(datasetId, apiToken);
+        datasetAsJson.prettyPrint();
+        String protocol = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.protocol");
+        String authority = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.authority");
+        String identifier = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.identifier");
+        String datasetPersistentId = protocol + ":" + authority + "/" + identifier;
+
+        // check thumbnails are empty
+
+        Response thumbnailCandidatesResponse = UtilIT.showDatasetThumbnailCandidates(datasetPersistentId, apiToken);
+        thumbnailCandidatesResponse.prettyPrint();
+        thumbnailCandidatesResponse.then().assertThat().statusCode(OK.getStatusCode());
+        List<Object> images = JsonPath.from(thumbnailCandidatesResponse.getBody().asString()).getList("data");
+        assertTrue(images.size() == 0);
+
+        // upload image files ( 1 good and 1 bad )
+
+        Response uploadResponse = UtilIT.uploadFileViaNative(datasetId.toString(), goodTiff, apiToken);
+        uploadResponse.then().assertThat().statusCode(OK.getStatusCode());
+        uploadResponse = UtilIT.uploadFileViaNative(datasetId.toString(), badTiff, apiToken);
+        uploadResponse.then().assertThat().statusCode(OK.getStatusCode());
+
+        // check thumbnails only contains 1 good image
+        thumbnailCandidatesResponse = UtilIT.showDatasetThumbnailCandidates(datasetPersistentId, apiToken);
+        thumbnailCandidatesResponse.prettyPrint();
+        thumbnailCandidatesResponse.then().assertThat().statusCode(OK.getStatusCode());
+        images = JsonPath.from(thumbnailCandidatesResponse.getBody().asString()).getList("data");
+        assertTrue(images.size() == 1);
+
+        // test set logo with badTiff and tiff to large (goodTiff)
+        Response uploadLogoResponse = UtilIT.uploadDatasetLogo(datasetPersistentId, badTiff, apiToken);
+        uploadLogoResponse.prettyPrint();
+        uploadLogoResponse.then().assertThat().statusCode(FORBIDDEN.getStatusCode());
+        uploadLogoResponse.then().assertThat().body("message", equalTo("In setNonDatasetFileAsThumbnail could not generate thumbnail from uploaded file."));
+
+        uploadLogoResponse = UtilIT.uploadDatasetLogo(datasetPersistentId, goodTiff, apiToken);
+        uploadLogoResponse.prettyPrint();
+        uploadLogoResponse.then().assertThat().statusCode(FORBIDDEN.getStatusCode());
+        uploadLogoResponse.then().assertThat().body("message", equalTo("File is larger than maximum size: 500000."));
```

Comment:
FWIW: Including the size makes the test dependent on the default max size.

Reply:
fixed